smarts/core/waypoints.py

```diff
@@ -300,6 +305,31 @@ def _equally_spaced_path(self, path):
             ref_waypoints_coordinates["ref_wp_headings"]
         )
 
+        first_wp_heading = ref_waypoints_coordinates["ref_wp_headings"][0]
+
+        wp_position = np.array([*path[0].wp.pos, 0])
+
+        vehicle_position = np.array([point[0], point[1], 0])
+
+        heading_vector = np.array(
+            [
+                math.cos(math.pi / 2 + first_wp_heading),
+                math.sin(math.pi / 2 + first_wp_heading),
+                0,
+            ]
+        )
+
+        projected_distant_wp_vehicle = np.inner(
+            (vehicle_position - wp_position), heading_vector
+        )
+
+        ref_waypoints_coordinates["ref_wp_positions_x"][0] = wp_position[
+            0
+        ] + projected_distant_wp_vehicle * math.cos(math.pi / 2 + first_wp_heading)
```

Comment:
Use the heading vector from above:
`wp_pos[0] + proj_dist * heading_vector[0]`

Reply:
Thanks, sure.